src/Evolution/Systems/GeneralizedHarmonic/GhDampingFunctions/GhDampingFunction.hpp

```diff
@@ -0,0 +1,63 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines base-class GhDampingFunction.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+/// \ingroup GhDampingFunctionsGroup
+/// Holds classes implementing GhDampingFunction (functions \f$R^n \to R\f$).
+namespace GhDampingFunctions {
+template <size_t VolumeDim, typename Fr>
+class GaussianPlusConstant;
+}  // namespace GhDampingFunctions
+
+/*!
+ * \ingroup GhDampingFunctionsGroup
+ * Encodes a function \f$R^n \to R\f$ where n is `VolumeDim` that represents
+ * a generalized-harmonic constraint-damping parameter (i.e., Gamma0,
+ * Gamma1, or Gamma2).
+ */
+template <size_t VolumeDim, typename Fr>
+class GhDampingFunction;
+
+/*!
+ * \ingroup GhDampingFunctionsGroup
+ * Encodes a function \f$R^n \to R\f$ where n is `VolumeDim` and where the
+ * function input (i.e., the spatial coordinates) is given as a rank-1 tensor.
+ */
+template <size_t VolumeDim, typename Fr>
+class GhDampingFunction : public PUP::able {
```

Comment:
We haven't been super consistent on this, but I find it easier if the base class is also in the namespace. I'd do:
```cpp
namespace GeneralizedHarmonic::ConstraintDamping {
template <...>
class DampingFunction : ...
```